modules/aws_ecs_fargate/main.tf

```diff
@@ -0,0 +1,247 @@
+  password                     = aws_secretsmanager_secret_version.rds_password.secret_string
+  port                         = 5432
+  publicly_accessible          = var.rds_publicly_accessible
+  vpc_security_group_ids       = [aws_security_group.rds.id]
+  performance_insights_enabled = var.rds_performance_insights_enabled
+  
+  skip_final_snapshot          = true
+  apply_immediately           = true
+}
+
+resource "aws_ecs_service" "retool" {
+  name                               = "${var.deployment_name}-main-service"
+  cluster                            = aws_ecs_cluster.this.id
+  task_definition                    = aws_ecs_task_definition.retool.arn
+  desired_count                      = var.ecs_service_count
+  deployment_maximum_percent         = var.maximum_percent
+  deployment_minimum_healthy_percent = var.minimum_healthy_percent
+  launch_type                        = "FARGATE"
+
+  load_balancer {
+    target_group_arn = aws_lb_target_group.this.arn
+    container_name   = "retool"
+    container_port   = 3000
+  }
+
+  network_configuration {
+    subnets         = var.subnet_ids
+    security_groups = [aws_security_group.alb.id]
+  }
+}
+
+resource "aws_eip" "retool_ip" {
+  vpc = true
+}
+
+resource "aws_nat_gateway" "retool_nat_gateway" {
+  allocation_id = aws_eip.retool_ip.id
+  subnet_id = var.public_subnet
+
+  tags = {
+    "Name" = "retool-nat"
+  }
+
+  depends_on = [
+    var.internet_gateway
+  ]
+}
+
+resource "aws_ecs_task_definition" "retool" {
+    family                   = "${var.deployment_name}-backend"
+    requires_compatibilities = ["FARGATE"]
+    network_mode             = var.ecs_task_network_mode
+    cpu                      = var.ecs_task_cpu
+    memory                   = var.ecs_task_memory
+    task_role_arn            = aws_iam_role.task_role.arn
+    execution_role_arn       = aws_iam_role.execution_role.arn
+    container_definitions    = <<TASK_DEFINITION
+[
+  {
+    "command": ["./docker_scripts/start_api.sh"],
+    "environment": [
+      {"name": "NODE_ENV", "value": "${var.node_env}"},
+      {"name": "SERVICE_TYPE", "value": "MAIN_BACKEND,DB_CONNECTOR"},
+      {"name": "FORCE_DEPLOYMENT", "value": "${var.force_deployment}"},
+      {"name": "POSTGRES_DB", "value": "hammerhead_production"},
+      {"name": "POSTGRES_HOST", "value": "${aws_db_instance.this.address}"},
+      {"name": "POSTGRES_SSL_ENABLED", "value": "${var.postgresql_ssl_enabled}"},
+      {"name": "POSTGRES_PORT", "value": "${var.postgresql_db_port}"},
+      {"name": "POSTGRES_USER", "value": "${aws_secretsmanager_secret_version.rds_username.secret_string}"},
+      {"name": "POSTGRES_PASSWORD", "value": "${aws_secretsmanager_secret_version.rds_password.secret_string}"},
+      {"name": "JWT_SECRET", "value": "${random_string.jwt_secret.result}"},
+      {"name": "ENCRYPTION_KEY", "value": "${random_string.encryption_key.result}"},
```

Comment:
i always am weary about randomly generating keys like this because if a customer were to want to keep their db but for some reason re-create their retool containers, they may get a new encryption key generated and nothing will work right :')

typically this stuff should be customer-provided and referenced through aws secrets manager in this case